### ISSUE TITLE
Render Images from Data Sources - Bracket Notation

### ISSUE THREAD

**Issue by alexalexson143** [FreeCodeCamp/FreeCodeCamp#12031]
Challenge [Render Images from Data Sources](https://www.freecodecamp.com/challenges/render-images-from-data-sources#?solution=%0Afccss%0A%20%20%24(document).ready(function()%20%7B%0A%0A%20%20%20%20%24(%22%23getMessage%22).on(%22click%22%2C%20function()%20%7B%0A%E2%80%89%20%20%20%20%20%24.getJSON(%22%2Fjson%2Fcats.json%22%2C%20function(json)%20%7B%0A%0A%20%20%20%20%20%20%20%20var%20html%20%3D%20%22%22%3B%0A%0A%20%20%20%20%20%20%20%20json.forEach(function(val)%20%7B%0A%20%20%20%20%20%20%20%20%20%20html%20%2B%3D%20%22%3Cdiv%20class%20%3D%20'cat'%3E%22%3B%0A%0A%20%20%20%20%20%20%20%20%20%20%2F%2F%20Only%20change%20code%20below%20this%20line.%0A%20%20%20%20%20%20%20%20%20%20html%20%2B%3D%20%22%3Cimg%20src%20%3D%20'%22%20%2B%20val%5B%22imageLink%22%5D%20%2B%20%22'%3E%22%3B%0A%20%20%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%20%20%2F%2F%20Only%20change%20code%20above%20this%20line.%0A%0A%20%20%20%20%20%20%20%20%20%20html%20%2B%3D%20%22%3C%2Fdiv%3E%22%3B%0A%0A%20%20%20%20%20%20%20%20%7D)%3B%0A%0A%E2%80%89%E2%80%89%20%20%E2%80%89%20%20%20%24(%22.message%22).html(html)%3B%0A%0A%20%20%20%20%20%20%7D)%3B%0A%20%20%20%20%7D)%3B%0A%20%20%7D)%3B%0Afcces%0A%0A%3Cdiv%20class%3D%22container-fluid%22%3E%0A%20%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%20%3Ch2%3ECat%20Photo%20Finder%3C%2Fh2%3E%0A%20%20%3C%2Fdiv%3E%0A%20%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%20well%20message%22%3E%0A%20%20%20%20%20%20The%20message%20will%20go%20here%0A%20%20%20%20%3C%2Fdiv%3E%0A%20%20%3C%2Fdiv%3E%0A%20%20%3Cdiv%20class%20%3D%20%22row%20text-center%22%3E%0A%20%20%20%20%3Cdiv%20class%20%3D%20%22col-xs-12%22%3E%0A%20%20%20%20%20%20%3Cbutton%20id%20%3D%20%22getMessage%22%20class%20%3D%20%22btn%20btn-primary%22%3E%0A%20%20%20%20%20%20%20%20Get%20Message%0A%20%20%20%20%20%20%3C%2Fbutton%3E%0A%20%20%20%20%3C%2Fdiv%3E%0A%20%20%3C%2Fdiv%3E%0A%3C%2Fdiv%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```html

<script>
  $(document).ready(function() {

    $("#getMessage").on("click", function() {
      $.getJSON("/json/cats.json", function(json) {

        var html = "";

        json.forEach(function(val) {
          html += "<div class = 'cat'>";

          // Only change code below this line.
          html += "<img src = '" + val["imageLink"] + "'>";
          
          // Only change code above this line.

          html += "</div>";

        });

        $(".message").html(html);

      });
    });
  });
</script>

<div class="container-fluid">
  <div class = "row text-center">
    <h2>Cat Photo Finder</h2>
  </div>
  <div class = "row text-center">
    <div class = "col-xs-12 well message">
      The message will go here
    </div>
  </div>
  <div class = "row text-center">
    <div class = "col-xs-12">
      <button id = "getMessage" class = "btn btn-primary">
        Get Message
      </button>
    </div>
  </div>
</div>

```

Forgive me if I don't know the correct terms. Bracket notation (val["imageLink]") not accepted as a correct solution even though clicking Get Message does the same thing with Dot Notation (val.imageLink).

**Comment by ajain17** [comment]
This seems to be working val[imageLink]. I think it needs to be fixed. The correct way of getting value of a key should be using quotes. Take a look at this fiddle for explanation:
https://jsfiddle.net/ro73j30s/1/

I think we need to get it fixed. Can someone else confirm it?

**Comment by raisedadead** [comment]
_help needed in triage._
/cc @FreeCodeCamp/moderators

**Comment by dhcodes** [comment]
This is an issue with the [test](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/580436380683fa762ed70f6e8e51c2e1ef263161/seed/challenges/01-front-end-development-certification/json-apis-and-ajax.json#L370)

As @ajain17 points out `val[imageLink]` does work, but only as a side effect of not escaping the `.` in the Regex test, thereby letting any value replace the `[`. The test needs to be re-written to strictly search for dot notation and also allow bracket notation with required quotes as in: `val["imageLink"]`.

**Comment by Manish-Giri** [comment]
@dhcodes @raisedadead I'd like to give this a shot!

**Comment by systimotic** [comment]
@Manish-Giri How's progress on this?

**Comment by Manish-Giri** [comment]
@systimotic My bad - just getting to this, been working on some of my other PRs, and then the break...

**Comment by systimotic** [comment]
@Manish-Giri No problem! Thanks for the PR!
